editors/vscode/client/extension.ts

```diff
@@ -331,15 +331,25 @@ function updateStatsBar(
 }
 
 function generateActivatorByConfig(config: VSCodeConfig, context: ExtensionContext): void {
-  const watcher = workspace.createFileSystemWatcher('**/.oxlintrc.json', false, true, true);
+  const watcher = workspace.createFileSystemWatcher('**/.oxlintrc.json', false, true, !config.requireConfig);
```

Comment:
This function never gets called when `config.requireConfig` is `false`
```suggestion
  const watcher = workspace.createFileSystemWatcher('**/.oxlintrc.json', false, true, false);
```